discovery-provider/src/tasks/index_user_listening_history.py

```diff
@@ -0,0 +1,137 @@
+import logging
+import time
+from typing import DefaultDict
+import sqlalchemy as sa
+from src.models.models import UserListeningHistory
+from src.tasks.celery_app import celery
+from src.models import IndexingCheckpoints, Play
+from src.utils.helpers import UPDATE_INDEXING_CHECKPOINTS_QUERY
+
+logger = logging.getLogger(__name__)
+
+USER_LISTENING_HISTORY_TABLE_NAME = "user_listening_history"
+BATCH_SIZE = 100000 # index 100k plays at most at a time
```

Comment:
this seems like a lot of records. do you think it'll cause any memory issues?

Reply:
Local testing seemed fine and completed in ~10 secs per batch but I can set this to be 10k instead. This really only matters when first populating.